functions/dataManger/gameManager.ts

```diff
@@ -0,0 +1,176 @@
+const AWS = require('aws-sdk');
+const isTest = process.env.JEST_WORKER_ID;
+const config = {
+  convertEmptyValues: true,
+  ...(isTest && {
+    endpoint: 'localhost:8000',
+    sslEnabled: false,
+    region: 'local-env',
+  }),
+};
+const docClient = new AWS.DynamoDB.DocumentClient(config);
+const table = (isTest) ? process.env.DYNAMO_DB_TEST_TABLE : process.env.DYNAMO_DB_GAME_TABLE;
+import { Game } from "../models/game";
+import { GameError } from "../error/gameErrorHandler";
+import { getPriceData } from "./priceDataManager";
+import * as Interfaces from "../interfaces/interfaces";
+
+  export async function createGame(game: Game): Promise<Game> {
+    try {
+      let params = {
+        TableName: table,
+        Item: {
+          partitionKey: game.partitionKey,
+          sortKey: game.sortKey,
+          gameName: game.gameName,          
+          genre: game.genre,
+          yearReleased: game.yearReleased,
+          developer: game.developer,
+          console: game.console,
+          desiredCondition: game.desiredCondition,
+          desiredPrice: game.desiredPrice,
+          priceData: (game.desiredPrice) ? await getPriceData(game) : undefined
+        },
+        ConditionExpression: 'attribute_not_exists(partitionKey) AND attribute_not_exists(sortKey)'
+      }
+      let response = await docClient.put(params).promise();
+      let createdGame = await getGame(game);             
+      return game;
+    } catch(err: any) {
+      throw new GameError(err.message, err.statusCode);
+    }
+  }
+
+ export async function getGame(game: Game) : Promise<Game> {
+    let params = {
+      TableName: table,
+      Key: {
+        partitionKey: game.partitionKey,
+        sortKey: game.sortKey
+      },
+      KeyConditionExpression: `sortKey = ${game.sortKey} AND partitionKey = ${game.partitionKey}`
+    }
+    
+    try {
+      let response = await docClient.get(params).promise();
+      let game = serializeDynamoResponse(response.Item);
+      return game;
+    } catch (err: any) {
+      //Dynamo returns an empty object if the get can't find a record.
+      //Not sure how to handle this since the documentClient doesn't throw an error
+      if (err.message == "Cannot read property 'partitionKey' of undefined") {
```

Comment:
I do still feel like there has to be a better way to handle this. Generally messages are considered volatile and not safe for conditional logic in code. I'd think about catching this prior to attempting to deserialize the response. I even think its ok to just return an empty object without an error (just like Dyanmo)